generators/server/index.js

```diff
@@ -154,6 +154,11 @@ module.exports = JhipsterServerGenerator.extend({
                 this.messageBroker = false;
             }
 
+            this.apiFirst = this.config.get('apiFirst') === 'no' ? false : this.config.get('apiFirst');
```

Comment:
that might be more complex so for the moment lets just have it as boolean similar to searchEngine option. If we ever add another apiFirst stuff we could reconsider or just add another boolean. In any case booleans are leaner and easy to maintain